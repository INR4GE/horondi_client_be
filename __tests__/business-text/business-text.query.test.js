/* eslint-disable no-undef */
const { gql } = require('@apollo/client');
const client = require('../../utils/apollo-test-client');
require('dotenv').config();
const {
  BUSINESS_TEXT_NOT_FOUND,
} = require('../../error-messages/business-text.messages');
const {
  newBusinessText,
  notExistBusinessTextId,
} = require('./business-text.variables');

let businessText = null;

describe('Business page queries', () => {
  beforeAll(async () => {
    const res = await client
      .mutate({
        mutation: gql`
          mutation($businessText: BusinessTextInput!) {
            addBusinessText(businessText: $businessText) {
              ... on BusinessText {
                _id
                code
                title {
                  value
                  lang
                }
                text {
                  value
                  lang
                }
              }
              ... on Error {
                message
                statusCode
              }
            }
          }
        `,
        variables: {
          businessText: newBusinessText,
        },
      })
      .then(response => response)
      .catch(e => e);

    businessText = res.data.addBusinessText;
  });

  afterAll(async () => {
    await client.mutate({
      mutation: gql`
        mutation($id: ID!) {
          deleteBusinessText(id: $id) {
            ... on BusinessText {
              _id
              code
              title {
                value
                lang
              }
              text {
                value
                lang
              }
            }
            ... on Error {
              message
              statusCode
            }
          }
        }
      `,
      variables: { id: businessText._id },
    });
  });

  test('#1 Should receive all business texts', async () => {
    const res = await client
      .query({
        query: gql`
          query {
            getAllBusinessTexts {
              code
              title {
                value
                lang
              }
              text {
                lang
                value
              }
            }
          }
        `,
      })
      .catch(e => e);

    const allTexts = res.data.getAllBusinessTexts;
    const justAddedText = allTexts[allTexts.length - 1];

    expect(allTexts).toBeDefined();
    expect(justAddedText).toEqual({
      __typename: 'BusinessText',
      title: newBusinessText.title,
      code: newBusinessText.code,
      text: newBusinessText.text,
    });
  });

  test('#2 Should receive selected business text', async () => {
    try {
      const res = await client
        .query({
          query: gql`
            query($id: ID!) {
              getBusinessTextById(id: $id) {
                ... on BusinessText {
                  code
                  title {
                    value
                    lang
                  }
                  text {
                    lang
                    value
                  }
                }
                ... on Error {
                  statusCode
                  message
                }
              }
            }
          `,
          variables: { id: businessText._id },
        })
        .catch(e => e);

      businessText = res.data.getBusinessTextById;

      expect(businessText).toBeDefined();
      expect(businessText).toHaveProperty('code', newBusinessText.code);
      expect(businessText.title).toBeInstanceOf(Array);
      expect(businessText).toHaveProperty('title', newBusinessText.title);
      expect(businessText.text).toBeInstanceOf(Array);
      expect(businessText).toHaveProperty('text', newBusinessText.text);
    } catch (e) {
      console.error(e);
    }
  });

  test('#3 Returning not existing business text should return error message', async () => {
    const res = await client
      .query({
        query: gql`
          query($id: ID!) {
            getBusinessTextById(id: $id) {
              ... on BusinessText {
                code
                title {
                  value
                  lang
                }
                text {
                  lang
                  value
                }
              }
              ... on Error {
                statusCode
                message
              }
            }
          }
        `,
        variables: { id: notExistBusinessTextId },
      })
      .catch(e => e);
    expect(res.data.getBusinessTextById).toBeDefined();
    expect(res.data.getBusinessTextById).toHaveProperty('statusCode', 404);
    expect(res.data.getBusinessTextById).toHaveProperty(
      'message',
      BUSINESS_TEXT_NOT_FOUND
    );
  });

  test('#4 Should receive selected business text by code', async () => {
    try {
      const res = await client
        .query({
          query: gql`
            query($code: String!) {
              getBusinessTextByCode(code: $code) {
                ... on BusinessText {
                  code
                  title {
                    value
                    lang
                  }
                  text {
                    lang
                    value
                  }
                }
                ... on Error {
                  statusCode
                  message
                }
              }
            }
          `,
          variables: { code: 'new-code' },
        })
        .catch(e => e);

      businessText = res.data.getBusinessTextByCode;

      expect(businessText).toBeDefined();
      expect(businessText).toHaveProperty('code', newBusinessText.code);
      expect(businessText.title).toBeInstanceOf(Array);
      expect(businessText).toHaveProperty('title', newBusinessText.title);
      expect(businessText.text).toBeInstanceOf(Array);
      expect(businessText).toHaveProperty('text', newBusinessText.text);
    } catch (e) {
      console.error(e);
    }
  });

  test('#5 Should return error if page by code not found', async () => {
    try {
      const res = await client
        .query({
          query: gql`
            query($code: String!) {
              getBusinessTextByCode(code: $code) {
                ... on BusinessText {
                  code
                  title {
                    value
                    lang
                  }
                  text {
                    lang
                    value
                  }
                }
                ... on Error {
                  statusCode
                  message
                }
              }
            }
          `,
          variables: { code: 'not-existing-code' },
        })
        .catch(e => e);

      expect(res.data.getBusinessTextByCode).toBeDefined();
      expect(res.data.getBusinessTextByCode).toHaveProperty('statusCode', 404);
      expect(res.data.getBusinessTextByCode).toHaveProperty(
        'message',
        BUSINESS_TEXT_NOT_FOUND
      );
    } catch (e) {
      console.error(e);
    }
  });
});
