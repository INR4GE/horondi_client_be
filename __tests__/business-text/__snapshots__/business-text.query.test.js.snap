// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business page queries  Should receive all business texts 1`] = `
Array [
  Object {
    "code": "about-us",
    "text": Array [
      Object {
        "lang": "uk",
        "value": "<p><span style=\\"font-size: large;\\">РЮКЗАКИ ЯК ТЕРАПІЯ</span></p>
<p><span style=\\"font-size: large;\\">HORONDI</span></p>
<p><span style=\\"font-size: large;\\">Сашко Горонді</span></p>

<p>Привіт! Я Олександр Горонді – засновник бренду <strong>HORONDI</strong>.
  Народився та виріс на Закарпатті і по правді кажучи - перші 20 років мого життя були далеко не самим
  приємними для спогадів чи розповідей. У мене не було на кого рівнятись, від кого черпати натхнення
  та отримувати знання про навколишній світ, як жити в ньому і по яких правилах. Тому, не маючи
  початково закладеного підґрунту правильних моральний та духовних цінностей - я був підлітком,
  який легко потрапляє під вплив. Хоч я і робив погані речі в житті, поганою людиною я себе не вважав.
  Робив все те, що робили люди кругом, мовчки і без зайвих запитань.</p>

<p><img src={ABOUT_US_IMAGES.horondi_1} alt='Oleksandr Horondi' /></p>

<p>Проте, десь у глибині ще жив той справжній я, якого доводилось приховувати, щоб мати змогу існувати
  в середовищі, де я зростав. Я завжди хотів вірити, що десь у світі для кожного з нас є особливе місце.
  Місце, де кожний з нас може наповнитись барвами життя, розквітнути як весняна квітка розцвітає під
  теплими проміннями ранішнього сонця, приносити радість оточуючим тебе людям. І, якби сталось так,
  що тим першим «вчителем», якого я зустрів на своєму шляху був би хтось куди більш доброзичливий
  тоді хтозна як все склалось б.</p>

<p><img src={ABOUT_US_IMAGES.horondi_2} alt='Oleksandr Horondi' /></p>

<p>Але сталось як сталось і я вдячний цьому, бо ця, хоч і терниста, стежка привела мене сюди.
  Шлях був довгий, але це історія зі щасливим кінцем і мені вдалось віднайти себе і зайнятись власним
  ремеслом. Це справа, яка дарує радість людям навколо. Справа – заради якої я просинаюсь зранку.
  Справа, яка мотивує мене рухатись вперед.</p>

<p><img src={ABOUT_US_IMAGES.horondi_3} alt='Oleksandr Horondi' /></p>

<p>Можливо, не варто було писати цього всього вище, так як не виключено, що ціль Вашого візиту просто
  купити річ, яка буде служити Вам багато років і дарувати радість.
  Але <strong>HORONDI</strong> - це не просто речі, які пошиті десь далеко
  за кордоном автоматизованими станками для шиття. <strong>HORONDI</strong> -
  це дещо більше. Кожна річ, будь це наплічник, гаманець чи сумочка – всі вони наділені душею,
  кожній з них я віддаю частинку своєї любові і в такий спосіб маю можливість розділяти її з людьми.
  В світі немає нічого благороднішого для мене ніж можливість викликати щиру посмішку на обличчях
  оточуючих і бачити радість в їхніх очах. Тому, купуючи <strong>HORONDI</strong> –
  Ви даєте мені можливість робити це.</p>

<p><span style=\\"font-size: large;\\">Команда</span></p>

<p><img src={ABOUT_US_IMAGES.workPlace_1} alt='Horondi team' /></p>

<p>Наша команда складається з п‘яти осіб, де кожен вірить в те, що робить. Ми працюємо від зорі до зорі,
  щоб Ви могли вчасно отримати свій <strong>HORONDI</strong>. Але при цьому ми
  не забуваємо про якість, яку Ви також заслуговуєте. Вся наша продукція пошита з якісної та приємної
  на дотик тканини з додатковим прошитим шаром дубляжу, що додає міцності та водостійкості. Дно зроблене
  з міцної тканини Кордура, шкіри чи еко-шкіри.</p>

<p><img src={ABOUT_US_IMAGES.workPlace_2} alt='Horondi team' /></p>

<p><strong>HORONDI</strong> розташований у Львові і місце, де ми створюємо нашу продукцію
  завбільшки, як звичайна кімната у Вашому домі. Тому, зараз ми не можемо похвалитись масштабами нашого підприємства,
  але ми ще молоді і тільки робимо перші кроки в цьому напрямку і з Вашою допомогою та підтримкою у нас все вийде!</p>
",
      },
      Object {
        "lang": "en",
        "value": "<p><span style=\\"font-size: large;\\">BACKPACKS AS THERAPY</span></p>
<p><span style=\\"font-size: large;\\">HORONDI</span></p>
<p><span style=\\"font-size: large;\\">Sashko Horondi</span></p>

<p>Hi there! I'm Oleksandr Horondi, founder of the <strong>HORONDI</strong>
    brand. I was born and grew up in Zakarpattia Oblast and, truth be told, the early 20 years of my life
    aren't always pleasant to recall or speak about. I had no one to look up to or be inspired by;
    no one taught me about the world, how to live in it and by what set of rules. Without any initial
    foundation of moral and spiritual values, I was an easily influenced teenager. Still, despite
    having done things I'm not proud of, I didn't consider myself a bad person. I just followed
    the lead of people around me, no questions asked.</p>

<p><img src={ABOUT_US_IMAGES.horondi_1} alt='Oleksandr Horondi' /></p>

<p>However, somewhere deep inside, a real me was hiding to survive the environment I was growing up in.
    I've always wanted to believe that each of us has a special place in the world. A place where one
    can be filled with all colors of life, blossom like a spring flower blooms under the warm beams
    of the morning son, and bring joy to the people around. If the first «teacher» I met in my path
    were a much kinder one, who knows how it all would have turned out.</p>

<p><img src={ABOUT_US_IMAGES.horondi_2} alt='Oleksandr Horondi' /></p>

<p>Be that as it may, I am grateful, for this road, bumpy as it was, has lead me here. It was a long
    journey, but this story has a happy ending, as I've managed to find myself and my craft. This pursuit
    brings joy to the people around me. It gets me out of bed in the morning. It motivates me to keep improving.</p>

<p><img src={ABOUT_US_IMAGES.horondi_3} alt='Oleksandr Horondi' /></p>

<p>Maybe I shouldn't have written all of that, as it's likely that you came here simply to buy an item
    that will serve you and bring joy for years to come. However, <strong>HORONDI</strong>
    is not machine-made products from somewhere far abroad. <strong>HORONDI</strong>
    is something more. Every item, be it a backpack, a wallet, or a bag, contains a sliver of my soul I've put
    in there to share it with people. There is nothing more precious to me in the whole world than an opportunity
    to bring a genuine smile to the faces of people around me and see joy in their eyes. That is why,
    when you buy <strong>HORONDI</strong>, you give me this opportunity.</p>

<p><span style=\\"font-size: large;\\">Team</span></p>

<p><img src={ABOUT_US_IMAGES.workPlace_1} alt='Horondi team' /></p>

<p>Our team consists of five people, all of whom believe in what we do. We work from dawn till dusk for you
    to receive your <strong>HORONDI</strong> in time. In addition, we make sure you receive
    the high quality you deserve. Our entire stock is made from high-quality fabric that is pleasant to the touch
    and has an additional durable and water-resistant layer. The bottom can be made from durable Cordura, leather or eco leather.</p>

<p><img src={ABOUT_US_IMAGES.workPlace_2} alt='Horondi team' /></p>

<p><strong>HORONDI</strong> is located in Lviv, and the space where we create our products
    is as large as a usual room in your place. Therefore, we can't boast the scale of our business yet.
    But we are still young and only making our first steps. With your help and support, we can do anything!</p>
",
      },
    ],
    "title": Array [
      Object {
        "lang": "uk",
        "value": "Про нас",
      },
      Object {
        "lang": "en",
        "value": "About Us",
      },
    ],
  },
  Object {
    "code": "terms-and-conditions",
    "text": Array [
      Object {
        "lang": "uk",
        "value": "Умови очікуйте найближчим часом...",
      },
      Object {
        "lang": "en",
        "value": "Terms and Conditions will appear shortly...",
      },
    ],
    "title": Array [
      Object {
        "lang": "uk",
        "value": "Умови",
      },
      Object {
        "lang": "en",
        "value": "Terms and Conditions",
      },
    ],
  },
  Object {
    "code": "payment-and-shipping",
    "text": Array [
      Object {
        "lang": "uk",
        "value": "Оплату і доставку очікуйте найближчим часом...",
      },
      Object {
        "lang": "en",
        "value": "Payment and Shipping will appear shortly...",
      },
    ],
    "title": Array [
      Object {
        "lang": "uk",
        "value": "Оплата і доставка",
      },
      Object {
        "lang": "en",
        "value": "Payment and Shipping",
      },
    ],
  },
  Object {
    "code": "contacts",
    "text": Array [
      Object {
        "lang": "uk",
        "value": "Контакти очікуйте найближчим часом...",
      },
      Object {
        "lang": "en",
        "value": "Contacts will appear shortly...",
      },
    ],
    "title": Array [
      Object {
        "lang": "uk",
        "value": "Контакти",
      },
      Object {
        "lang": "en",
        "value": "Contacts",
      },
    ],
  },
  Object {
    "code": "new-code",
    "text": Array [
      Object {
        "lang": "uk",
        "value": "Тут бізнес текст",
      },
      Object {
        "lang": "en",
        "value": "Business text here",
      },
    ],
    "title": Array [
      Object {
        "lang": "uk",
        "value": "НоваБТ",
      },
      Object {
        "lang": "en",
        "value": "NewBT",
      },
    ],
  },
]
`;

exports[`Business page queries  Should receive selected business text 1`] = `
Object {
  "code": "new-code",
  "text": Array [
    Object {
      "lang": "uk",
      "value": "Тут бізнес текст",
    },
    Object {
      "lang": "en",
      "value": "Business text here",
    },
  ],
  "title": Array [
    Object {
      "lang": "uk",
      "value": "НоваБТ",
    },
    Object {
      "lang": "en",
      "value": "NewBT",
    },
  ],
}
`;
